ggml/src/ggml-cpu/unary-ops.cpp

```diff
@@ -122,65 +131,80 @@ static void unary_op(const ggml_compute_params * params, ggml_tensor * dst) {
 }
 
 void ggml_compute_forward_abs(const ggml_compute_params * params, ggml_tensor * dst) {
-    unary_op<op_abs>(params, dst);
+    unary_op(op_abs, params, dst);
 }
 
 void ggml_compute_forward_sgn(const ggml_compute_params * params, ggml_tensor * dst) {
-    unary_op<op_sgn>(params, dst);
+    unary_op(op_sgn, params, dst);
 }
 
 void ggml_compute_forward_neg(const ggml_compute_params * params, ggml_tensor * dst) {
-    unary_op<op_neg>(params, dst);
+    unary_op(op_neg, params, dst);
 }
 
 void ggml_compute_forward_step(const ggml_compute_params * params, ggml_tensor * dst) {
-    unary_op<op_step>(params, dst);
+    unary_op(op_step, params, dst);
 }
 
 void ggml_compute_forward_tanh(const ggml_compute_params * params, ggml_tensor * dst) {
-    unary_op<op_tanh>(params, dst);
+    unary_op(op_tanh, params, dst);
 }
 
 void ggml_compute_forward_elu(const ggml_compute_params * params, ggml_tensor * dst) {
-    unary_op<op_elu>(params, dst);
+    unary_op(op_elu, params, dst);
 }
 
 void ggml_compute_forward_relu(const ggml_compute_params * params, ggml_tensor * dst) {
-    unary_op<op_relu>(params, dst);
+    unary_op(op_relu, params, dst);
 }
 
 void ggml_compute_forward_sigmoid(const ggml_compute_params * params, ggml_tensor * dst) {
-    unary_op<op_sigmoid>(params, dst);
+    unary_op(op_sigmoid, params, dst);
 }
 
 void ggml_compute_forward_hardsigmoid(const ggml_compute_params * params, ggml_tensor * dst) {
-    unary_op<op_hardsigmoid>(params, dst);
+    unary_op(op_hardsigmoid, params, dst);
 }
 
 void ggml_compute_forward_exp(const ggml_compute_params * params, ggml_tensor * dst) {
-    unary_op<op_exp>(params, dst);
+    unary_op(op_exp, params, dst);
 }
 
 void ggml_compute_forward_hardswish(const ggml_compute_params * params, ggml_tensor * dst) {
-    unary_op<op_hardswish>(params, dst);
+    unary_op(op_hardswish, params, dst);
 }
 
 void ggml_compute_forward_sqr(const ggml_compute_params * params, ggml_tensor * dst) {
-    unary_op<op_sqr>(params, dst);
+    unary_op(op_sqr, params, dst);
 }
 
 void ggml_compute_forward_sqrt(const ggml_compute_params * params, ggml_tensor * dst) {
-    unary_op<op_sqrt>(params, dst);
+    unary_op(op_sqrt, params, dst);
 }
 
 void ggml_compute_forward_sin(const ggml_compute_params * params, ggml_tensor * dst) {
-    unary_op<op_sin>(params, dst);
+    unary_op(op_sin, params, dst);
 }
 
 void ggml_compute_forward_cos(const ggml_compute_params * params, ggml_tensor * dst) {
-    unary_op<op_cos>(params, dst);
+    unary_op(op_cos, params, dst);
 }
 
 void ggml_compute_forward_log(const ggml_compute_params * params, ggml_tensor * dst) {
-    unary_op<op_log>(params, dst);
+    unary_op(op_log, params, dst);
 }
+
+void ggml_compute_forward_xielu(const ggml_compute_params * params, ggml_tensor * dst) {
+    // Get the XIELU parameters from the operation
+    float alpha_n = ggml_get_op_params_f32(dst, 1);
+    float alpha_p = ggml_get_op_params_f32(dst, 2);
```

Comment:
Why do you read these values as `alpha_n` and `alpha_p` here when what is stored is something else?